boot-mongodb-elasticsearch/src/main/java/com/example/mongoes/web/controller/RestaurantController.java

```diff
@@ -26,79 +27,77 @@
 @RestController
 @Timed
 @RequestMapping("/api/restaurant")
-public class RestaurantController {
+@Validated
+class RestaurantController {
 
     private final RestaurantService restaurantService;
 
-    public RestaurantController(RestaurantService restaurantService) {
+    RestaurantController(RestaurantService restaurantService) {
         this.restaurantService = restaurantService;
     }
 
     @GetMapping
-    public Mono<ResponseEntity<SearchPage<Restaurant>>> findAllRestaurants(
-            @Valid @RequestParam(defaultValue = "10") @Size(max = 999) int limit,
+    Mono<ResponseEntity<SearchPage<Restaurant>>> findAllRestaurants(
+            @Valid @RequestParam(defaultValue = "10") @Max(999) int limit,
```

Comment:
@coderabbitai, please help me to write junit which validates the limit parameter

Reply:
@coderabbitai, how to define global exception handler for Validation. Current code is throwing 500 exception instead of 400 bad request